src/js/components/Navigation/TabWithPushHistory.jsx

```diff
@@ -1,22 +1,45 @@
 import { Tab } from '@mui/material';
-import PropTypes from 'prop-types';
-import React from 'react';
+import TagManager from 'react-gtm-module';
 import { Link } from 'react-router-dom'; // useHistory
-
-
+import React from 'react';
+import PropTypes from 'prop-types';
+import lookupPageNameAndPageTypeDict, { getPageDetails } from '../../utils/lookupPageNameAndPageTypeDict';
+import VoterStore from '../../stores/VoterStore';
 /*
  The simplest way to define a component is to write a JavaScript function:
  This function is a valid React component because it accepts a single “props”
  (which stands for properties) object argument with data and returns a React element.
  We call such components “function components” because they are literally JavaScript functions.
  https://reactrouter.com/native/api/Hooks/usehistory
 */
+
+// DataLayer helper function
+function pushDataLayer (destinationPath, buttonId = '') {
+  const destinationPage = lookupPageNameAndPageTypeDict(destinationPath);
+  const dataLayerObject = {
+    actionDetails: {
+      actionType: 'navigate',
+      buttonId,
+    },
+    event: 'click',
```

Comment:
Since this is an internal event (instead of going to another website), this should be:
```
event: 'action',
```